src/kOS.Safe/Encapsulation/ScalarValue.cs

```diff
@@ -402,10 +407,16 @@ object IConvertible.ToType(Type conversionType, IFormatProvider provider)
             // ignoring the logic used in Create() to vary the type depending on content.
             if (conversionType == GetType())
                 return this;  // no conversion needed
+            else if (conversionType == typeof(ScalarValue))
+                return this; // no conversion needed
             else if (conversionType == typeof(ScalarDoubleValue))
                 return new ScalarDoubleValue(GetDoubleValue());
             else if (conversionType == typeof(ScalarIntValue))
                 return new ScalarIntValue(GetIntValue());
+            else if (conversionType == typeof(BooleanValue))
+                return new BooleanValue(GetIntValue() == 0 ? false : true);
+            else if (conversionType.IsSubclassOf(typeof(Structure)))
+                throw new KOSCastException(typeof(ScalarValue), conversionType);
```

Comment:
I wish it was possible to do this as a `switch` statement because they're faster than an if/else ladder, but they require a numeric or enumerated conditional value in order to perform the magic "jump right to the correct line" work, and I don't think GetType() works that way.  I think it's '==' checks are not simple reference equals checks.  So we'll have to live with this.